week4/week4.py

```diff
@@ -0,0 +1,61 @@
+# week4
+from collections import defaultdict
+
+
+class LargestSumRectangle:
+    def __init__(self, grid: list):
+        self.grid = grid
+        self.rows = len(grid)
+
+    # pre-compute row sums starting at every index ( 0(n*m) size )
+    def getRowSums(self) -> dict:
+        grid = self.grid
+        rows = self.rows
+        sums = defaultdict(int)
+
+        for i in range(rows):
+            s = 0
+            for j in range(len(grid[i])):
+                try:
+                    s += grid[i][j]
+                    sums[(i,j)] = s
+                except:
+                    s = 0
+        return sums
+
+    # returns a tuple of pairs of indicies, the first pair representing the index of left top corner of the rectangle
+    #  and the second pair representing the index of right bottom corner of the rectangle. 
+    # algorithm runs in 0(n^2 * m). 
```

Comment:
Prefer to have these as docstrings, they go directly after the method declaration in triple quotes.